### PR TITLE
Enhancements/pyroscan apply func

### PR DESCRIPTION
**Pull request by bpatel2107**
Added in documentation on how to perform parameter scans.

I've also snuck in an upgrade on the `PyroScan` object allowing you to apply functions when running scans, along with a test

### CODE REVIEW

File: docs/howtos/parameter_scans.rst [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+    )
+
+    # Write input files
+    pyro_scan.write()
+
+This will write 3 different input files where the parameter :py:attr:`pyrokinetics.numerics.Numerics.ky` has been
+changed, with each input file in its own directory under a base directory of ``run_directory``.
+
+The variable here is always defined in pyrokinetics default units, so when using a code that has a different normalised be
+aware that the values defined here will be converted to the code own units
```
Comment (mgiacomin) [inline docs/howtos/parameter_scans.rst]: Perhaps you could point here to the documentation where pyro units are explained.

Comment (bpatel2107) [reply docs/howtos/parameter_scans.rst]: Currently writing that now but will when its done!

File: docs/howtos/parameter_scans.rst [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+        parameter_separator="_",
+        base_directory="run_directory"
+    )
+
+After which the process is the same. Note that an outer product is formed of all the specified parameters so the number
+of runs can become large very quickly.
+
+Here we have specified:
+
+How to separate each value from its parameter: ``value_separator``
```
Comment (mgiacomin) [inline docs/howtos/parameter_scans.rst]: Maybe you can add bullets where you list "here we have specified".

**Review by mgiacomin — APPROVED** [review]
Looks good.

File: docs/howtos/parameter_scans.rst (hunk 1 above)
Comment (bpatel2107) [reply docs/howtos/parameter_scans.rst]: Done

**Pull request merged**